src/controls/scale.test.ts

```diff
@@ -0,0 +1,75 @@
+import { Canvas, Point, Rect, type RectProps } from 'fabric';
+import { scalingX, scalingY } from './scale';
+
+//
+// prep for tests
+//
+const createZeroThickRectangleScalingItems = (
+  rectOptions: { width: number; height: number } & Partial<RectProps>,
+  pointDiff: Point
+) => {
+  // create rect
+  const target = new Rect({
+    strokeWidth: 0,
+    stroke: 'black',
+    ...rectOptions,
+  });
+  // add rect to canvas or we get undefined issues
+  const canvas = new Canvas('canvas', { width: 100, height: 100 });
+  canvas.add(target);
+
+  // create mouse event near center of rect, as the 0 size will put it on the middle scaler
+  const canvasOffset = canvas.calcOffset();
+  const moveEvent = new MouseEvent('mousemove', {
+    clientX: canvasOffset.left + target.left + target.width / 2 - pointDiff.x,
+    clientY: canvasOffset.top + target.top + target.height / 2 - pointDiff.y,
+  });
+  canvas.setActiveObject(target);
+  target.__corner = target._findTargetCorner(
+    canvas.getViewportPoint(moveEvent)
+  );
+  canvas._setupCurrentTransform(moveEvent, target, false);
```

Comment:
```suggestion
  canvas._setupCurrentTransform(moveEvent, target, true);
```

Don't you want to control in the transform?

Reply:
That's what I get for copying the first test from setupCurentTransform. Thanks!